### PR TITLE
Fix override of custom validation rules starting with 'unique'

### PR DESCRIPTION
**Pull request by Flynsarmy**
Creating a custom validation rule starting with *unique* and adding it to your model results in unintended behaviour when updating the model, as your custom role is overridden with the default *unique* rule.

For example:
```php
Validator::extendDependent('unique_with', function (string $attribute, $value, array $parameters, \Illuminate\Validation\Validator $validator) {
    dd('hi');
});

class MyModel extends Model
{
    use \October\Rain\Database\Traits\Validation;

    public $table = 'my_table';
    public $rules = [
        'key' => 'unique_with:foo,bar',
    ];
}
```
The `dd()` will never execute with this rule on your model, because these lines 
https://github.com/octobercms/library/blob/bb6d88a0f8b33b535de71101ea281245ebe7267c/src/Database/Traits/Validation.php#L379-L381
replace our rule with
> unique:my_table,1,id

### CODE REVIEW

**Comment by bennothommo** [comment]
@Flynsarmy you could just use a name that does not start with "unique" :)

**Comment by Flynsarmy** [comment]
I could, but we really shouldn't leave issues like this just because we can. Took me ages to figure out why October was acting so strangely. 

This bug also potentially breaks third party rules brought in in packages. My inspiration was in fact a unique_with package I found on github

File: src/Database/Traits/Validation.php [hunk 1]
```diff
@@ -376,7 +376,7 @@ protected function processValidationRules($rules)
                 /*
                  * Remove primary key unique validation rule if the model already exists
                  */
-                if (starts_with($rulePart, 'unique') && $this->exists) {
+                if (($rulePart == 'unique' || starts_with($rulePart, 'unique:')) && $this->exists) {
```
Comment (LukeTowers) [inline src/Database/Traits/Validation.php]:
```suggestion
                if (($rulePart === 'unique' || starts_with($rulePart, 'unique:')) && $this->exists) {
```

**Comment by bennothommo** [comment]
@Flynsarmy fair points.

**Pull request merged**